app/Http/Controllers/API/EventController.php

```diff
@@ -319,29 +354,416 @@ private function getUser() {
+     *                ),
+     *                @OA\Property(
+     *                   property="end",
+     *                   ref="#/components/schemas/Event/properties/start",
+     *                ),
+     *                @OA\Property(
+     *                   property="timezone",
+     *                   ref="#/components/schemas/Event/properties/timezone",
+     *                ),
+     *                @OA\Property(
+     *                   property="title",
+     *                   ref="#/components/schemas/Event/properties/title",
+     *                ),
+     *                @OA\Property(
+     *                   property="description",
+     *                   ref="#/components/schemas/Event/properties/description",
+     *                ),
+     *                @OA\Property(
+     *                   property="location",
+     *                   ref="#/components/schemas/Event/properties/location",
+     *                ),
+     *                @OA\Property(
+     *                   property="online",
+     *                   ref="#/components/schemas/Event/properties/online",
+     *                ),
+     *             )
+     *         )
+     *    ),
+     *    @OA\Response(
+     *        response=200,
+     *        description="Successful operation",
+     *        @OA\JsonContent(
+     *            @OA\Property(
+     *              property="data",
+     *              title="data",
+     *              ref="#/components/schemas/Event"
+     *            )
+     *        ),
+     *     )
+     *  )
+     */
+    public function updateEventv2(Request $request, $idEvents) {
+        $user = $this->getUser();
+
+        list($groupid, $start, $end, $title, $description, $location, $timezone, $latitude, $longitude, $online, $link) = $this->validateEventParams(
+            $request,
+            false
+        );
+
+        // Convert the timezone to UTC, because the timezone is not itself stored in the DB.
+        $event_start_utc = Carbon::parse($start)->setTimezone('UTC')->toIso8601String();
+        $event_end_utc = Carbon::parse($end)->setTimezone('UTC')->toIso8601String();
+        $hours = Carbon::parse($event_start_utc)->diffInHours(Carbon::parse($event_end_utc));
+
+        // We don't let the user change the group that an event is on.
+        $update = [
+            'event_start_utc' => $event_start_utc,
+            'event_end_utc' => $event_end_utc,
+            'hours' => $hours,
+            'free_text' => $description,
+            'online' => $online,
+            'venue' => $title,
+            'link' => $link,
+            'location' => $location,
+            'latitude' => $latitude,
+            'longitude' => $longitude,
+            'timezone' => $timezone
+        ];
+
+        $party = Party::findOrFail($idEvents);
+        $party->update($update);
+
+        // REVIEW: In the old code we wouldn't generate EditEvent in the approval case.  But I think that was a bug.
```

Comment:
I'm not sure.  Because $party->approve() triggers ApproveEvent, which creates a post on WordPress.  So if we're then triggering EditEvent immediately afterwards, we'd be updating the just created event.  Which might be harmless enough, but I guess there's also a possibility of a race condition if they're queued (which I can't remember if they are)

Reply:
They are queued, though I would expect that won't cause a race because we have a single queue worker.  There is a window where we will approve on Wordpress and then edit, but I don't think we need to worry too much about that because the Wordpress site should (and will) be using the API to access the data.

Conceptually, I think the point of events is to indicate what has happened, and it is then up to the handler to decide what to do with them.  We shouldn't suppress generation of an event when the thing has happened as a way of stopping the handler doing something, otherwise we lose the point of the interface.